bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
+
+        self.dropdown: discord.ui.Select = self.children[0]
+        for duration in self.SNOOZE_DURATIONS:
+            self.dropdown.add_option(label=duration)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """Ensure that the user clicking the button is the member who invoked the command."""
+        if interaction.user.id != self.interaction_owner_id:
+            await interaction.response.send_message(":x: This is not your reminder to snooze!", ephemeral=True)
+            return False
+        return True
+
+    @discord.ui.select(placeholder="Select the snooze duration")
+    async def select_snooze_duration(self, _: discord.ui.Select, interaction: discord.Interaction) -> None:
+        """Drop down menu which contains a list of snooze durations one can choose."""
+        selected_duration = interaction.data["values"][0]
+        current_datetime = datetime.now()
+        if (num_seconds := self.SNOOZE_DURATIONS.get(selected_duration)) is None:
+            amount, unit = selected_duration.split()
+            if unit.startswith("month"):
+                num_seconds = (current_datetime + relativedelta(months=int(amount)) - current_datetime).total_seconds()
+
+            else:  # unit is years
+                num_seconds = (current_datetime + relativedelta(years=int(amount)) - current_datetime).total_seconds()
+
+        new_reminder_end_time = current_datetime + relativedelta(seconds=num_seconds)
+
+        await self.bot.api_client.patch(
+            f'bot/reminders/{self.reminder_id}',
+            json={'expiration': new_reminder_end_time.isoformat()}
+        )
+
+        self.new_expiry = new_reminder_end_time
+        new_end_timestamp = time.discord_timestamp(new_reminder_end_time, format=time.TimestampFormats.DAY_TIME)
+        original_embed = interaction.message.embeds[0]
+        new_embed = discord.Embed(
+            colour=discord.Colour.green(),
+            title=random.choice(POSITIVE_REPLIES),
+            description=f"Successfully snoozed reminder for {selected_duration}. New remind time: {new_end_timestamp}",
+        )
+        new_embed.set_footer(text=f"ID: {self.reminder_id}")
+        try:
+            await interaction.response.edit_message(embeds=[original_embed, new_embed], view=None)
+        except discord.NotFound:
+            await interaction.message.channel.send(embed=new_embed)
+
+        self.reminder_was_snoozed = True
+        self.stop()
+
+
+class SnoozeButtonView(discord.ui.View):
+    """The reminder's snooze button UI View."""
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=300)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.reminder_author_id = reminder_author_id
+
+        snooze_button = SnoozeButton(self)
+        self.add_item(snooze_button)
```

Comment:
Just a minor nit pick, why not have
```py
snooze_button = SnoozeButton(self)
self.add_item(snooze_button)
```
as
```py
self.add_item(SnoozeButton(self))
```
just seems like an unnecessary variable assignment but please let me know if there's a reason for this.

Reply:
Yep, agreed!